src/app/products/charges/create-charge/create-charge.component.ts

```diff
@@ -0,0 +1,230 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, FormControl, Validators } from '@angular/forms';
+import { Router, ActivatedRoute } from '@angular/router';
+import { DatePipe } from '@angular/common';
+
+/** Custom Services */
+import { ProductsService } from '../../products.service';
+
+/**
+ * Create charge component.
+ */
+@Component({
+  selector: 'mifosx-create-charge',
+  templateUrl: './create-charge.component.html',
+  styleUrls: ['./create-charge.component.scss']
+})
+export class CreateChargeComponent implements OnInit {
+
+  /** Charge form. */
+  chargeForm: FormGroup;
+  /** Charges template data. */
+  chargesTemplateData: any;
+  /** Charge time type data. */
+  chargeTimeTypeData: any;
+  /** Charge calculation type data. */
+  chargeCalculationTypeData: any = '';
+  /** Income and liability account data */
+  incomeAndLiabilityAccountData: any;
+  /** Minimum due date allowed. */
+  minDate = new Date(2000, 0, 1);
+  /** Maximum due date allowed. */
+  maxDate = new Date(new Date().setFullYear(new Date().getFullYear() + 1));
+  /** Repeat every label */
+  repeatEveryLabel: string;
+  /** Currency decimal points */
+  currencyDecimalPoints: number;
+
+  /**
+   * Retrieves the charges template data and income and liability account data from `resolve`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {ProductsService} productsService Products Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   * @param {DatePipe} datePipe Date Pipe to format date.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private productsService: ProductsService,
+              private route: ActivatedRoute,
+              private router: Router,
+              private datePipe: DatePipe) {
+    this.route.data.subscribe((data: { chargesTemplate: any }) => {
+      this.chargesTemplateData = data.chargesTemplate;
+      this.incomeAndLiabilityAccountData = data.chargesTemplate.incomeOrLiabilityAccountOptions.incomeAccountOptions
+        .concat(data.chargesTemplate.incomeOrLiabilityAccountOptions.liabilityAccountOptions);
+    });
+  }
+
+  /**
+   * Creates and sets the charge form.
+   */
+  ngOnInit() {
+    this.createChargeForm();
+    this.setChargeForm();
+    this.setConditionalControls();
+  }
+
+  /**
+   * Creates the charge form.
+   */
+  createChargeForm() {
+    this.chargeForm = this.formBuilder.group({
+      'chargeAppliesTo': ['', Validators.required],
+      'name': ['', Validators.required],
+      'currencyCode': ['', Validators.required],
+      'chargeTimeType': ['', Validators.required],
+      'chargeCalculationType': ['', Validators.required],
+      'amount': ['', [Validators.required, Validators.pattern('^\\s*(?=.*[1-9])\\d*(?:\\.\\d+)?\\s*$')]],
```

Comment:
I am able to input `1.` with no number after the decimal place and it does not show any error. Check this.

Reply:
I have found the issue, it is to do with the type="number", which converts '1.' to '1.0' automatically (I have not been able to find a way to change this at all). I think I will remove type="number" as it is having many problems integrating with angular validators. I will instead add a (keyup) event instead which will allow only numbers and decimal places to be inputted. Is this fine by you?